test/unit/sites.js

```diff
@@ -3,179 +3,325 @@
+        done();
+      });
+    });
+  });
+
+  describe('getSite(req)', function() {
+    it('get a site', function(done) {
+      var promise = sites.getSite({
+        params: {
+          site_id: 1
+        }
+      });
+
+      promise.then(function(data) {
+        assert.deepEqual([site1], data);
         done();
       });
     });
 
-    xit('get a site', function(done) {
+    it('gets a site that does not exist', function(done) {
       var promise = sites.getSite({
         params: {
-          site_id: 12345
+          site_id: 124
         }
       });
 
       promise.then(function(data) {
-        assert.equal({
-          site_id: 12345,
-          site_name: 'single',
-          site_address: 'single Boston, MA'
-        }, data[0]);
+        assert.deepEqual([], data);
         done();
       });
     });
+  });
 
-    xit('updates a site', function(done) {
-      site.getSite({
+  describe('updateSite(req)', function() {
+    it('updates a site', function(done) {
+      sites.getSite({
         params: {
-          site_id: 54321
+          site_id: 9
         }
       })
       .then(function(data) {
-        assert.equal({
-          site_id: 54321,
-          site_name: 'old name',
-          site_address: 'old address'
-        }, data[0]);
+        assert.deepEqual([site9], data);
 
         return sites.updateSite({
           params: {
-            site_id: 54321
+            site_id: 9
           },
-          data: {
-            updateValues: {
-              site_name: 'new name',
-              site_address: 'new address'
-            }
+          body: {
+            site_name: 'new name',
+            site_address: 'new address'
           }
         });
       })
       .then(function() {
-        return site.getSite({
+        return sites.getSite({
           params: {
-            site_id: 54321
+            site_id: 9
           }
         });
       })
       .then(function(data) {
-        assert.equal({
-          site_id: 54321,
+        assert.deepEqual([{
+          site_id: 9,
           site_name: 'new name',
           site_address: 'new address'
-        }, data[0]);
+        }], data);
         done();
       });
     });
 
-    xit('delete a site', function(done) {
+    it('does not updated a site because the body was missing', function(done) {
+      sites.updateSite({}).catch(function(err) {
+        assert.equal(err.status, 406);
+        assert.equal(err.message, 'Must provide site\'s name, or address');
+        done();
+      });
+    });
+
+    it('does not updated a site because there are no fields to be updated', function(done) {
+      sites.updateSite({
+        body: {}
+      })
+      .catch(function(err) {
+        assert.equal(err.status, 406);
+        assert.equal(err.message, 'Must provide site\'s name, or address');
+        done();
+      });
+    });
+
+    it('updates a site that does not exist', function(done) {
```

Comment:
More clear "It does nothing when asked to update a site that doesn't exist" or something